File: py/atm.py
balance = 5000
Card_No = int(input("Welcome to UBA Bank. Enter card number: "))
pwd = int(input("Enter PIN to be saved: "))

print("Saving...")
print("Saved")
print("===============================================================================================================")
print("Welcome once more to United Bank of Africa, Cameroon branch. Select from the list below to perform your action.")

while True:
    print(
        '''
        1.) Withdraw
        2.) Send
        3.) Check balance
        4.) Exit
        '''
    )
    choice = int(input("Please enter your choice: "))

    if choice == 1:
        amount = float(input("Enter amount: "))

        if amount > balance:
            print("Insufficient balance.")
        else:
            for attempt in range(3):
                pwd1 = int(input("Enter your password: "))
                if pwd1 == pwd:
                    balance -= amount
                    print(f"You have successfully withdrawn FCFA {amount} from your account. Your new balance is FCFA {balance}")
                    break
                else:
                    print("Wrong password. Try again")
            else:
                print("You have entered the wrong password multiple times. Try again later")
                break
            
    elif choice == 2:
        amount = float(input("Enter amount: "))
        accNumber = int(input("Enter receiver's account number: "))
        print(f"Checking for account number '{accNumber}' in the database...")
        if amount > balance:
            print("Insufficient balance.")
        else:
            for attempt in range(3):
                pwd1 = int(input("Enter your password: "))
                if pwd1 == pwd:
                    balance -= amount
                    print(f"You have successfully sent FCFA {amount} to account '{accNumber}'. Your new balance is FCFA {balance}")
                    break
                else:
                    print("Wrong password. Try again")
            else:
                print("You have entered the wrong password multiple times. Try again later")
                
    elif choice == 3:
        for attempt in range(3):
            pwd1 = int(input("Enter your password: "))
            if pwd1 == pwd:
                print(f"Your balance is FCFA {balance}")
                break
            else:
                print("Wrong password. Try again")
        else:
            print("You have entered the wrong password multiple times. Try again later")

    elif choice == 4:
        print("Thanks for coming through")
        break
    
    else:
        print("Invalid choice. Enter a valid input next time")

    response = input("Do you want to perform another transaction? (yes/no): ")
    if response.lower() != "yes":
        print("Thanks for using UBA Bank. Have a great day!")
        break

File: cpp/atm.cpp
// This is an ATM machine in cpp
#include <iostream>
#include <string>

using namespace std;

int main () {
int cardNo, pin;
int Cardno, pinn;
int choice;
int amount;
int initialBalance = 500;
string answer;
cout << "Good day. Enter card No to be saved: ";
cin >> cardNo;
cout << "Enter pin to be saved: ";
cin >> pin;
cout << "Saving..." << endl;

system("cls");

while (true) {

cout << "Welcome to this ATM service. Enter card No to continue: ";
cin >> Cardno;
if (Cardno == cardNo) {
    cout << "1. Deposit\n";
    cout << "2. Withdraw\n";
    cout << "3. Balance\n";
    cout << "4. Exit\n";
    cout << "Please enter desired option: ";
    cin >> choice;

    switch (choice) {
        case 1: 
            cout << "Enter amount: ";
            cin >> amount;
            for (int i = 0; i < 3; i++) {
            cout << "Enter your pin: ";
            cin >> pinn;
                if (pin == pinn) {
                    initialBalance += amount;
                    cout << "You have succesfully deposited " << amount << " FCFA to your account. ";
                    cout << "Your new balance is " << initialBalance << "FCFA" <<endl;
                    break;
                }
                else 
                    {
                        cout << "Invalid PIN. Try again." << endl;
                    }
            }
            break;
        
        case 2: 
            cout << "Enter amount: ";
            cin >> amount;
            if (amount <= initialBalance) {
            for (int i = 0; i < 3; i++) {
            cout << "Enter your pin: ";
            cin >> pinn;
                if (pin == pinn) {
                    initialBalance -= amount;
                    cout << "You have succesfully withdrawn " << amount << " FCFA to your account. ";
                    cout << "Your new balance is " << initialBalance << "FCFA" <<endl;
                    break;
                }
                else 
                    {
                        cout << "Invalid PIN. Try again" << endl;
                    }
            }
            }
            else
                {
                    cout << "Insufficient funds"<<endl;
                    exit(1);
                }
            break;
            
        case 3:
            for (int i = 0; i < 3; i++) {
            cout << "Please enter your pin: ";
            cin >> pinn;
                if (pin == pinn) {
                    cout << "Your balance is " << initialBalance << " FCFA" << endl;
                }
                else
                    {
                        cout << "Wrong pin\n";
                        cout << "Try again"<<endl;
                    }
            }
            break;

        case 4:
            cout << "Thanks for using our service"<<endl;
            exit(1);
            break;

        default:
            cout << "Invalid option entered\n"; 
            break;
    }

}
else
    {
        cout << "Invalid card number\n";
    }

        cout << "Do you want to perform another transaction? (y/n): ";
        cin >> answer;
        if ((answer == "y") || (answer == "Y")) {
                cout << "----------------------------------------------------------------"<<endl;
                cout << "                                                                 "<<endl;
            continue;
        }
        else {
            exit(1);
        }
}
system("pause");
}



